cmd/juju/cloud/list.go

```diff
@@ -78,14 +108,41 @@ func (c *listCloudsCommand) SetFlags(f *gnuflag.FlagSet) {
 		"json":    cmd.FormatJson,
 		"tabular": formatCloudsTabular,
 	})
+	f.StringVar(&c.controllerName, "c", "", "Controller to operate in")
+	f.StringVar(&c.controllerName, "controller", "", "")
+}
+
+func (c *listCloudsCommand) getCloudList() (*cloudList, error) {
+	if c.controllerName == "" {
+		details, err := listCloudDetails()
+
+		if err != nil {
+			return nil, err
+		}
+		return details, nil
+	}
+
+	api, err := c.listCloudsAPIFunc()
+	if err != nil {
+		return nil, err
+	}
+	controllerClouds, err := api.Clouds()
+	if err != nil {
+		return nil, err
+	}
+	details := newCloudList()
+	for name, cloud := range controllerClouds {
+		cloudDetails := makeCloudDetails(cloud)
+		details.public[name.String()] = cloudDetails
```

Comment:
so here's the thing - they won't all be public
we don't really record from where they came - public, built in, personal
this will do for now but add a todo to see if we can categorise